omgeo/services/__init__.py

```diff
@@ -658,9 +658,11 @@ def _geocode(self, pq):
                 query = dict(query, searchExtent=pq.viewbox.to_esri_wgs_json())
         else: # single-line
             method = 'find'
+            magic_key = pq.key if hasattr(pq, 'key') else ''
```

Comment:
Indeed, this matches our findings, and we were seeing results from `/find` where we would get no results at all when we supplied text + magicKey + bbox + categories params. 

Is a sane solution here to remove all passed parameters other than text + magicKey if a magicKey is provided? While potentially helpful, I'm a bit weary because this behavior isn't documented by ESRI afaik, and its 'magic' to the user.

Or, we just add some helpful documentation indicating this issue and leave it up to the user to sanitize the arguments?